sky/shell/platform/mac/platform_mac.mm

```diff
@@ -53,6 +55,19 @@ int PlatformMacMain(int argc,
 
   InitializeLogging();
 
+  base::CommandLine& command_line = *base::CommandLine::ForCurrentProcess();
+  if (command_line.HasSwitch(sky::shell::switches::kTraceStartup)) {
+    // Usually, all tracing within flutter is managed via the tracing controller
+    // The tracing controller is accessed via the shell instance. This means
+    // that tracing can only be enabled once that instance is created. Traces
+    // early in startup are lost. This enables tracing in base manually till
+    // the tracing controller takes over.
+    base::trace_event::TraceLog::GetInstance()->SetEnabled(
+        base::trace_event::TraceConfig("*",
+                                       base::trace_event::RECORD_UNTIL_FULL),
+        base::trace_event::TraceLog::RECORDING_MODE);
+  }
```

Comment:
Can we move the body of this if-statement into platform-independent code so that we can call it from all the various platform's main functions?